src/Common/AwsClient.php

```diff
@@ -224,19 +222,21 @@ public function getPaginator($name, array $args = [], array $config = [])
             . "of {$this->api['serviceFullName']} cannot be paginated.");
     }
 
-    public function getWaiter($name, array $args = [], array $config = [])
+    public function waitUntil($name, array $args = [], array $config = [])
     {
+        // Prepare waiter config and args.
         $config += $this->api->getWaiterConfig($name);
+        $args += ['@future' => false];
```

Comment:
Is this used the waiter object somehow? If not, I think this could be removed and the later check for this attribute could be changed to `empty()`.